### PR TITLE
test(ESD-1385): add output formatter unit tests for read-only command packages

### PR DESCRIPTION
**Pull request by Phil-Browne**
Adds the missing `<pkg>_output_test.go` for the six read-only command packages (`partners`, `product`, `status`, `billing_market`, `auth`, `locations`). Each now exercises the `to<Type>Output` field mapping and all four output formats (table, JSON, CSV, XML), plus a nil/empty case. XML was previously untested for every one of these.

Two real bugs surfaced while writing the tests; fixed with regressions kept:

- Nil-safety: `toPartnerOutput`, `toBillingMarketOutput`, `toLocationOutput`, `toLocationTableOutput` dereferenced their pointer args without a nil check, and the partner/location/country/billing print loops appended raw entries. A nil element from the API would panic. Added nil guards.
- Status XML element names: the status dashboard uses a bespoke XML encoder and the output structs had no `xml` tags, so it emitted Go field names (`<UID>`, `<ASN>`) instead of the snake_case names every other command and the dashboard's own JSON use. Added `xml` tags so XML matches (`<uid>`, `<asn>`).

### CODE REVIEW

File: internal/commands/billing_market/billing_market_output_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+func TestBillingMarketOutput_EmptySlice(t *testing.T) {
+	for _, format := range []string{"table", "json", "csv", "xml"} {
+		t.Run(format, func(t *testing.T) {
+			out := output.CaptureOutput(func() {
+				err := output.PrintOutput([]billingMarketOutput{}, format, true)
+				assert.NoError(t, err)
+			})
+			assert.NotPanics(t, func() { _ = out })
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/billing_market/billing_market_output_test.go]: Fixed in 99943fb. The empty-slice test now asserts the actual per-format output (header rows for table/csv, [] for json, empty items element for xml) instead of the no-op check.

File: internal/commands/billing_market/billing_market_actions.go [hunk 2]
```diff
@@ -38,6 +38,9 @@ func GetBillingMarkets(cmd *cobra.Command, args []string, noColor bool, outputFo
 
 	outputs := make([]billingMarketOutput, 0, len(markets))
 	for _, m := range markets {
+		if m == nil {
+			continue
+		}
 		outputs = append(outputs, toBillingMarketOutput(m))
 	}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/billing_market/billing_market_actions.go]: Good catch. Moved the warning to key off the filtered outputs slice (len(outputs) == 0), so an all-nil response now warns instead of printing empty output silently. Added a unit test covering the all-nil case. Checked the sibling output files (locations, partners) and their action emitters; none emit a "No X found" warning before filtering, so no change needed there.

File: internal/commands/status/status_output_test.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+	assert.Equal(t, dashboardSummary{Ports: 1, MCRs: 1, MVEs: 1, VXCs: 1, IXs: 1}, dashboard.Summary)
+}
+
+func TestPrintDashboard_Table(t *testing.T) {
+	dashboard := statusTestDashboard(t)
+	out := op.CaptureOutput(func() {
+		err := printDashboard(dashboard, "table", true)
+		assert.NoError(t, err)
+	})
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/status/status_output_test.go:75]: Good catch. printDashboard does not set the global output format (only StatusDashboard does), so these tests inherited whatever a prior test left. Added a withOutputFormat helper that pins and restores the global format per test; applied to all printDashboard tests. Pushed in ff62384.

File: internal/commands/status/status_output_test.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	assert.Contains(t, out, `"a_end_uid": "a-end"`)
+	assert.Contains(t, out, `"summary"`)
+}
+
+func TestPrintDashboard_CSV(t *testing.T) {
+	dashboard := statusTestDashboard(t)
+	out := op.CaptureOutput(func() {
+		err := printDashboard(dashboard, "csv", true)
+		assert.NoError(t, err)
+	})
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/status/status_output_test.go:105]: Fixed by the same withOutputFormat helper, which sets the global format to csv for this test and restores it afterward. Pushed in ff62384.

File: internal/commands/status/status_output_test.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+func TestPrintDashboard_Empty(t *testing.T) {
+	dashboard, err := buildDashboard(nil, nil, nil, nil, nil)
+	assert.NoError(t, err)
+
+	for _, format := range []string{"table", "json", "csv", "xml"} {
+		t.Run(format, func(t *testing.T) {
+			out := op.CaptureOutput(func() {
+				err := printDashboard(dashboard, format, true)
+				assert.NoError(t, err)
+			})
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/status/status_output_test.go:138]: Fixed: the subtest loop now calls withOutputFormat(t, format) per iteration, so each format subtest pins the global output format and CaptureOutput reliably sees stdout. Pushed in ff62384.

File: internal/commands/partners/partners_output.go [hunk 6]
```diff
@@ -31,6 +34,9 @@ func toPartnerOutput(p *megaport.PartnerMegaport) partnerOutput {
 var printPartnersFunc = func(partners []*megaport.PartnerMegaport, format string, noColor bool) error {
 	outputs := make([]partnerOutput, 0, len(partners))
 	for _, partner := range partners {
+		if partner == nil {
+			continue
+		}
 		outputs = append(outputs, toPartnerOutput(partner))
 	}
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/partners/partners_output.go:41]: Good catch, the filter ran before the print-time skip. Made filterPartners nil-safe by treating nil entries as non-matching, so they are dropped before FindPartners reaches printPartnersFunc. Added TestFilterPartners_NilElementsSkipped. Pushed in 93c2430.

File: internal/commands/locations/locations_output.go [hunk 7]
```diff
@@ -132,13 +141,19 @@ func printLocations(locations []*megaport.LocationV3, format string, noColor boo
 	if format == utils.FormatTable {
 		tableOutputs := make([]locationTableOutput, 0, len(locations))
 		for _, loc := range locations {
+			if loc == nil {
+				continue
+			}
 			tableOutputs = append(tableOutputs, toLocationTableOutput(loc))
```
[the remark this answers is not in the document]
Comment (Phil-Browne) [reply internal/commands/locations/locations_output.go:147]: Same fix on the locations path: filterLocations now skips nil entries before evaluating filters, so ListLocations no longer panics on a nil element. Added TestFilterLocations_NilElementsSkipped. Pushed in 93c2430.

**Review by penzeliz-megaport — APPROVED** [review]
Good formatter coverage across all four output formats, and the no-secret-leak test on auth status is a nice touch. Flagging for other reviewers that this also carries production changes, not just tests: nil-guards in the billing_market/locations/partners output paths, and a reorder in GetBillingMarkets so the "no markets" warning fires on the post-filter result rather than the raw input (an all-nil slice now warns, a partially-nil slice is filtered without warning). Both behaviours are covered by the new tests and look correct. Signing off the behaviour change with the tests. LGTM.
